test/gurka/test_route_incidents.cc

```diff
@@ -145,37 +150,82 @@ struct test_reader : public baldr::GraphReader {
   test_reader(const boost::property_tree::ptree& pt) : baldr::GraphReader(pt) {
     tile_extract_.reset(new baldr::GraphReader::tile_extract_t(pt));
   }
-  virtual std::shared_ptr<baldr::GraphReader::incident_tile_t>
+  virtual std::shared_ptr<valhalla::incidents::IncidentsTile>
   GetIncidentTile(const baldr::GraphId& tile_id) const {
     auto i = incidents.find(tile_id.Tile_Base());
     if (i == incidents.cend())
       return {};
     return std::shared_ptr<
-        baldr::GraphReader::incident_tile_t>(&i->second, [](baldr::GraphReader::incident_tile_t*) {});
+        valhalla::incidents::IncidentsTile>(&i->second, [](valhalla::incidents::IncidentsTile*) {});
   }
-  void add(const baldr::GraphId& id, baldr::GraphReader::Incident&& incident) {
-    incidents[id.Tile_Base()].emplace_back(incident);
+  void add(const baldr::GraphId& id,
+           valhalla::incidents::Location&& _incident_location,
+           uint64_t incident_id,
+           const std::string& incident_description = "") {
+
+    // Grab the incidents-tile in which we should insert both metadata and the edge-to-metadata
+    // relation
+    valhalla::incidents::IncidentsTile& incidents_tile = incidents[id.Tile_Base()];
+
+    uint32_t metadata_index = 0;
+    valhalla::incidents::Metadata* metadata = nullptr;
+    // Check if this incident is already tracked
+    for (auto& meta : *incidents_tile.mutable_metadata()) {
+      if (incident_id == meta.id()) {
+        // We've already added the incident, use it instead of adding again
+        metadata = &meta;
+        break;
+      }
+      metadata_index += 1;
+    }
```

Comment:
why was this necessary? just in case we had a typo in the tests and used the same id twice?